src/did.ts

```diff
@@ -0,0 +1,176 @@
+import * as base58 from 'base58-universal/main.js'
+import { CryptoSystem, Msg } from 'keystore-idb/types'
+
+import eccOperations from 'keystore-idb/ecc/operations'
+import rsaOperations from 'keystore-idb/rsa/operations'
+import utils from 'keystore-idb/utils'
+
+import * as dns from './dns'
+import * as keystore from './keystore'
+import { arrbufs } from './common'
+
+
+const ECC_DID_PREFIX: ArrayBuffer = new Uint8Array([ 0xed, 0x01 ]).buffer
+const RSA_DID_PREFIX: ArrayBuffer = new Uint8Array([ 0x00, 0xf5, 0x02 ]).buffer
+const BASE58_DID_PREFIX: string = 'did:key:z'
+
+
+
+// KINDS
+
+
+/**
+ * Create a DID to authenticate with.
+ */
+export async function local(): Promise<string> {
+  const ks = await keystore.get()
+  const pubKeyB64 = await ks.publicWriteKey()
+
+  return publicKeyToDid(pubKeyB64, ks.cfg.type)
+}
+
+/**
+ * Gets the root DID for a user.
+ * Stored at `_did.${username}.${domain}`
+ */
+export async function root(
+  username: string,
+  domain = 'fission.name'
+): Promise<string> {
+  try {
+    const maybeDid = await dns.lookupTxtRecord(`_did.${username}.${domain}`)
+    if (maybeDid !== null) return maybeDid
+  } catch (_err) {}
```

Comment:
🤨 

```ts
catch (_) {}
throw new Error(..)
```

Why not stick the `throw` inside the `catch`? Does this phrasing have a special kind of behaviour?

Reply:
Hmm not sure, just copied this from Daniel's PR.
I assume it's because of async/await, but not sure.
Thoughts @dholms ?